File: Algorithm/Programmers/Level 2/카펫.py
import math


def solution(brown, yellow):
    root = math.trunc(yellow ** (1 / 2))
    yaksu = []
    for i in range(1, root + 1):
        if yellow % i == 0:
            yaksu.append(yellow / i)
    yaksu.sort()
    for i in yaksu:
        if (i + 2) * ((yellow / i) + 2) == brown + yellow:
            return [i + 2, ((yellow / i) + 2)]

File: Algorithm/Programmers/Java/Level 2/카펫.java
public class 카펫 {
    public int[] solution(int brown, int yellow) {
        int size = brown + yellow;
        for(int i = 3; i < size; i++) {
            if(size % i != 0) {
                continue;
            }
            
            if((i - 2) * ((size / i) - 2) == yellow) {
                int[] answer = new int[2];
                answer[0] = Math.max(i, size / i);
                answer[1] = Math.min(i, size / i);
                return answer;
            }
        }
        return null;
    }
}
